flang/lib/Optimizer/Transforms/DebugTypeGenerator.cpp

```diff
@@ -583,6 +583,9 @@ DebugTypeGenerator::convertType(mlir::Type Ty, mlir::LLVM::DIFileAttr fileAttr,
     return convertVectorType(vecTy, fileAttr, scope, declOp);
   } else if (mlir::isa<mlir::NoneType>(Ty)) {
     return mlir::LLVM::DINullTypeAttr::get(context);
+  } else if (mlir::isa<mlir::IndexType>(Ty)) {
+    return genBasicType(context, mlir::StringAttr::get(context, "integer"), 64,
```

Comment:
Is IndexType 64 bits on all platforms? What about 32bit systems? 128 bit systems?

Reply:
I was thinking about the same issue and tried to see if this information is parameterized somewhere. But I found that `LLVMTypeConverter` [here](https://github.com/llvm/llvm-project/blob/98e3075df992636fa42aafde96748d1d5c834688/flang/lib/Optimizer/CodeGen/TypeConverter.cpp#L132) seems to always use the 64-bit for the index type so used the same logic.

Reply:
`getIndexTypeBitwidth()` in TypeConverter

Reply:
Thanks for the pointer. I have updated the PR to use it instead of a hardcoded 64.